### PR TITLE
[CARBONDATA-4232] Add missing doc change for secondary index.

### PR DESCRIPTION
**Pull request by nihal0107**
### Why is this PR needed?
 Documentation changes were not handled in PR [4116](https://github.com/apache/carbondata/pull/4116)
 
 ### What changes were proposed in this PR?
Added missing documentation.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by nihal0107** [comment]
retest this please.

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5626/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3883/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/30/

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5629/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3886/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/33/

**Comment by brijoobopanna** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3908/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5651/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/55/

**Comment by nihal0107** [comment]
retest this please.

File: docs/index/secondary-index-guide.md [hunk 1]
```diff
@@ -118,9 +118,11 @@ also load data to the SI table with the same number of segments as the main tabl
 the main table will also load data to the SI table.
 
  **NOTE**:
- * In case of data load failure to the SI table, then we make the SI table disable by setting a hive serde
- property. The subsequent main table load will load the old failed loads along with current load and
- makes the SI table enable and available for query.
+ * In case of data load failure to the SI table, segments of main table and SI table will not be in sync.
+ In this scenario the subsequent query pruning will be performed at segment level. It means all successful SI segment
```
Comment (Indhumathi27) [inline docs/index/secondary-index-guide.md:122]:
```suggestion
 In this scenario the subsequent query pruning will be performed at segment level. It means all successful SI segments 
```

Comment (nihal0107) [reply docs/index/secondary-index-guide.md]: done

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3916/

**Comment by nihal0107** [comment]
retest this please

File: docs/index/secondary-index-guide.md [hunk 2]
```diff
@@ -118,9 +118,11 @@ also load data to the SI table with the same number of segments as the main tabl
 the main table will also load data to the SI table.
 
  **NOTE**:
- * In case of data load failure to the SI table, then we make the SI table disable by setting a hive serde
- property. The subsequent main table load will load the old failed loads along with current load and
- makes the SI table enable and available for query.
+ * In case of data load failure to the SI table, segments of main table and SI table will not be in sync.
```
Comment (maheshrajus) [inline docs/index/secondary-index-guide.md:121]: can we give one example for this scenario for easy understanding ? 
ex:
segment 1 loaded successfully to maintable and SI
segment 2 loaded successfully to maintable and SI
segment 3 failed to load. 
show segment differences between maintable and SI table.
so in this case for segment 1 and segment 2 pruned by SI and segment 3 filter with help of maintable.

Comment (nihal0107) [reply docs/index/secondary-index-guide.md:121]: As discussed already examples present in the design doc which present in CARBONDATA-4162.

**Comment by maheshrajus** [comment]
LGTM

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5662/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/66/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3918/

**Comment by Indhumathi27** [comment]
LGTM

**Pull request closed without merging**